src/CMakeLists.txt

```diff
@@ -294,15 +289,39 @@ ELSE()
 	MESSAGE(FATAL_ERROR "Asked for unknown SHA1 backend ${SHA1_BACKEND}")
 ENDIF()
 
-# Include PCRE and its POSIX regex compatibility layer when it is required
-IF (HAVE_REGCOMP_L)
+# Specify regular expression implementation
+IF(REGEX STREQUAL "")
+	CHECK_SYMBOL_EXISTS(regcomp_l "regex.h;xlocale.h" HAVE_REGCOMP_L)
+	CHECK_SYMBOL_EXISTS(pcre_regcomp "pcreposix.h" HAVE_PCRE)
+
+	IF (HAVE_REGCOMP_L)
+		SET(REGEX "regcomp_l")
+	ELSEIF (HAVE_PCRE)
+		SET(REGEX "pcre")
+	ELSE ()
+		SET(REGEX "builtin")
+	ENDIF ()
```

Comment:
We have said in the past that there shall be no space between keyword and opening brace in conditionals